src/ripple/protocol/SOTemplate.h

```diff
@@ -35,6 +35,8 @@ enum SOEStyle {
     soeREQUIRED = 0,  // required
     soeOPTIONAL = 1,  // optional, may be present with default value
     soeDEFAULT = 2,   // optional, if present, must not have default value
+                      // inner object with the default fields has to be
+                      // constructed with STObject::makeInnerObject()
```

Comment:
I think this comment should copied to InnerObjectFormats.cpp, probably before line 138 (the closing `}`) or after line 27 (the opening `{`). That's where people are likely going to be making additions, and so will be more likely to see and heed the message.